### PR TITLE
Add token substitution for assets

### LINKED ISSUE #2404: Document token substitution for assets
### Description
We will ship in 5.20 support for token substitution in various runtime assets for the following resources:

- checks (agent entity)
- hooks (agent entity)
- handlers (backend entity)
- mutators (backend entity)

See [the issue](https://github.com/sensu/sensu-go/issues/3639) and [the PR](https://github.com/sensu/sensu-go/pull/3720).

The (only?) main use cases are using token substitution in the URL of an asset, so each backend and/or agent can download assets from the appropriate URL, or for customizing the headers. It's important to note that the SHA512 value does not accept token substitution, for security reasons. Also, token substitution cannot be used in the filters; the filters already have access to the entity so it's not a concern.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [X] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
- https://docs.sensu.io/sensu-go/latest/reference/assets/ ?
- https://docs.sensu.io/sensu-go/latest/reference/tokens/

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Add information about token substitution for assets to:
- https://docs.sensu.io/sensu-go/5.20/reference/tokens/
- https://docs.sensu.io/sensu-go/5.20/reference/assets/

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2404

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@palourde I based most of my explanation on clarifications added to the original issue here: https://github.com/sensu/sensu-docs/issues/2404#issuecomment-623566888
However, I don't know how close to the mark I am with this PR -- I don't think I have a good understanding of this feature.

Also, I think I should add an example that covers this usage to I think we should add an example to http://localhost:1313/sensu-go/5.20/reference/tokens/#token-substitution-syntax.

File: content/sensu-go/5.20/reference/tokens.md [hunk 1]
```diff
@@ -27,15 +28,30 @@ The agent replaces any tokens with matching attributes from the entity definitio
 Invalid templates or unmatched tokens return an error, which is logged and sent to the Sensu backend message transport.
 Checks with token-matching errors are not executed.
 
-Token substitution is supported for [check definition][7] `command` attributes and [hook][8] `command` attributes.
+Token substitution is supported for [check][7], [hook][8], [handler][10], and [mutator][11] `command` attributes.
```
Comment (palourde) [inline content/sensu-go/5.20/reference/tokens.md]: Token substitution is actually not supported for handlers & mutators; we now support token substitution for assets, which handlers & mutators can rely on, via the runtime_assets.

Here's a concrete example. Let's say I have this asset definition:

```
type: Asset
api_version: core/v2
metadata:
  name: sensu-go-hello-world
  namespace: default
spec:
  builds:
  - sha512: 06665fea5b7c7ee15e7822820aa7ddb791cc6338e38444e976e601bc7d7970592e806a7b88733690a238b7325437d31f85e98ae2fe47b008ca09c86530da9600
    url: "{{ .labels.asset_url }}/sensu-go-hello-world-0.0.1.tar.gz"
```

Now, checks & hooks can use the `sensu-go-hello-world` asset as a runtime_asset, and we will do token substitution with the agent's entity. The same things applies for assets referred by mutators & handlers, except we will use the backend's entity and not the agent's entity. The use case here is that you could host your assets on different URLs depending on the datacenter etc. without having to duplicate your assets.

File: content/sensu-go/5.20/reference/assets.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ If the downloaded artifact's SHA512 checksum matches the checksum provided by th
 Set the backend or agent's local cache path with the `--cache-dir` flag.
 Disable assets for an agent with the agent `--disable-assets` [configuration flag][30].
 
+Use [token substitution][3] for asset paths on disk to retrieve changes to assets without manually updating your resource commands.
```
Comment (palourde) [inline content/sensu-go/5.20/reference/assets.md]: I'm not sure to understand that bit. Right now the asset path on the filesystem is not done yet (https://github.com/sensu/sensu-go/issues/3662).

File: content/sensu-go/5.20/reference/tokens.md [hunk 3]
```diff
@@ -27,15 +28,30 @@ The agent replaces any tokens with matching attributes from the entity definitio
...
 ## Manage entity labels
 
 You can use token substitution with any defined [entity attributes][4], including custom labels.
 See the [entity reference][6] for information about managing entity labels for proxy entities and agent entities.
 
+## Manage asset changes
+
+Use token substitution to retrieve changes to assets without manually updating your check, hook, handler, and mutator commands.
+
+For example, if an asset changes, the asset's specific path on disk will change.
```
Comment (palourde) [inline content/sensu-go/5.20/reference/tokens.md]: Same as ^. Not sure if we should document the asset local path in a subsequent PR once we have the actual implementation.

**Review by palourde — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.20/reference/assets.md (hunk 2 above)
Comment (hillaryfraley) [reply content/sensu-go/5.20/reference/assets.md]: I'm sorry, I think we thought the whole asset templating thing would be part of 5.20. It sounds like that's incorrect, right?

File: content/sensu-go/5.20/reference/tokens.md (hunk 1 above)
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/tokens.md]:
```suggestion
Token substitution is supported for [check definition][7] `command` attributes and [hook][8] `command` attributes.
```

File: content/sensu-go/5.20/reference/tokens.md [hunk 4]
```diff
@@ -357,3 +373,5 @@ spec:
 [7]: ../checks/#check-commands
 [8]: ../hooks/
 [9]: #token-data-type-limitations
+[10]: ../handlers/
+[11]: ../mutators/
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/tokens.md:413]:
```suggestion
[11]: ../mutators/
[12]: ../assets/
```

File: content/sensu-go/5.20/reference/tokens.md [hunk 5]
```diff
@@ -29,13 +30,28 @@ Checks with token-matching errors are not executed.
 
 Token substitution is supported for [check definition][7] `command` attributes and [hook][8] `command` attributes.
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/tokens.md]:
```suggestion
Token substitution is supported for [check definition][7] `command` attributes and [hook][8] `command` attributes, as well as for [assets][12], which [handlers][10] and [mutators][11] can use via their `runtime_assets` attributes.
```

File: content/sensu-go/5.20/reference/tokens.md (hunk 3 above)
Comment (hillaryfraley) [reply content/sensu-go/5.20/reference/tokens.md]: yes, I think you are right if that will not be part of 5.20.

File: content/sensu-go/5.20/reference/tokens.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/5.20/reference/tokens.md]: Thank you -- this is really helpful. I will update the PR text and hopefully get it right.

File: content/sensu-go/5.20/reference/tokens.md [hunk 6]
```diff
@@ -27,15 +28,30 @@ The agent replaces any tokens with matching attributes from the entity definitio
...
 Available attributes will always have [string values][9], such as labels and annotations.
 
 ## Manage entity labels
 
 You can use token substitution with any defined [entity attributes][4], including custom labels.
 See the [entity reference][6] for information about managing entity labels for proxy entities and agent entities.
 
+## Manage asset changes
+
+Use token substitution to retrieve changes to assets without manually updating your check, hook, handler, and mutator commands.
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/tokens.md]:
```suggestion
Use token substitution in your [asset][12] definitions, which your [checks][7], [hooks][8], [handlers][10] and [mutators][11] can use via their `runtime_assets` attributes.
Token substitution in assets allows you to host your assets on different URLs (for example, at different datacenters) without duplicating your assets.
```

File: content/sensu-go/5.20/reference/assets.md (hunk 2 above)
Comment (palourde) [reply content/sensu-go/5.20/reference/assets.md]: That's right, I don't think the whole asset templating functionalities will be part of 5.20

**Review by palourde — APPROVED** [review]

**Pull request merged**